action.yml

```
@@ -329,6 +333,34 @@ runs:
                   '\n[Full pipeline log](${{ steps.artifacts_url.outputs.url }}/${{ steps.sched_test.outputs.req_id }}/pipeline.log).'
           })
 
+    - name: Create github summary
+      if: ${{ inputs.create_github_summary != 'false' }}
+      shell: bash
+      run: |
+        if [ -z "${{ steps.final_state.outputs.INFRA_STATE }}" ]; then
+          infra_state=OK
+        else
+          infra_state=Failed
+        fi
+        user_summary="${{ inputs.create_github_summary }}"
+        if [ "$user_summary" != 'true' ]; then
+          # In this case we expect key=value string inside
+          # 'create_github_summary' input.
+          # We obtain from it a table column and insert it to the GITHUB_STEP_SUMMARY.
+          user_key="${user_summary%%=*}|"
+          user_value="${user_summary##*=}|"
+          user_line="--|"
+        fi
+        tfaga_summary_header="Testing Farm as a GitHub Action summary:"
+        if  ! cat "$GITHUB_STEP_SUMMARY" | grep "$tfaga_summary_header" > /dev/null; then
```

```suggestion
        if  ! grep "$tfaga_summary_header" "$GITHUB_STEP_SUMMARY" > /dev/null; then
```

The  $GITHUB_STEP_SUMMARY is the environment variable and not a file.

---

E.g. I have defined variable $GITLAB_API_TOKEN.

```bash
$ cat "$GITLAB_API_TOKEN"
cat: asdfasdfasdfasdfasdf: No such file or directory
$
```

---

I think that in a `$GITHUB_STEP_SUMMARY` is stored name of a file where the summary is written.
We do not write to variables with >> or >, we use it to write to files.

---

You are right, based on the https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables